doc/tests.rst

```diff
@@ -448,8 +449,160 @@ which displays nicer looking diffs that the standard ``diff``
+``cat``. This is convenient for creating small fixtures.
+
+Directory Tests
+---------------
+
+In the above example we used ``cat`` to create the fixture, but what if there
+are too many fixtures to comfortably fit in test file? Or some of the fixtures
+are binary? It's possible to include the fixtures as normal files or directories
+provided the test is defined as a directory. The name of the test directory must
+end with ``.t`` and must include a ``run.t`` as the test script. Everything else
+in that directory is treated as raw data for the test. It's not possible to
+define rules using ``dune`` files in such a directory.
+
+We convert the ``wc`` test above into a directory test ``wc.t``:
+
+.. code:: bash
+
+   $ ls wc.t
+     run.t foo.txt bar/
+
+This defines a directory test ``wc.t`` which must include a ``run.t`` file as
+the test script, with ``fool.txt`` and ``bar`` are fixtures. We may then access
+their contents in the test script ``run.t``:
+
+.. code:: bash
+
+   $ wc -l foo | awk '{ print $1 }'
+   4
+   $ wc -l $(ls bar) | awk '{ print $1 }'
+   1231
+
+Test Options
+------------
+
+When testing binaries, it's important to to specify a dependency on the binary
+for two reasons:
+
+- Dune must know to re-run the test when a dependency changes
+
+- The dependencies must be specified to guarantee that they are visible to the
+  test when running it.
+
+We can specify dependencies using the ``deps`` field using the usual syntax:
+
+.. code:: bash
+
+   (cram
+    (deps ../foo.exe))
+
+This introduces a dependency on ``foo.exe`` on all cram tests in this directory.
+To apply the stanza to a particular test, it's possible to use ``applies_to``
+field:
+
+.. code:: bash
+
+   (cram
+    (applies_to * \ foo bar)
+    (deps ../foo.exe))
+
+We use the predicate language to apply this stanza to all tests in this
+directory except for ``foo.t`` and ``bar.t``. The ``applies_to`` field also
+accepts the value ``:whole_subtree`` in order to apply the options to all tests
+in all sub directories (recursively). This is useful to apply common options to
+an entire test suite.
+
+The ``cram`` stanza accepts the following fields:
+
+- ``enabled_if`` - controls whether the tests are enabled
+- ``alias`` - aliases that can be used to run the test (in addition to
+  ``runtest``)
+- ``deps`` - dependencies of the test
+
+A single test may be configured by more than one ``cram`` stanza. In such cases,
+the values from all applicable ``cram`` stanzas are merged together to get the
+final values for all the fields.
+
+Sandboxing
+----------
+
+Since cram tests often create intermediate artifacts, it's important that cram
+tests are executed in a clean environment. This is why all cram tests are
+sandboxed. To respect sandboxing, every test should specify dependency on any
+artifact that might rely on using the ``deps`` field.
+
+See :ref:`dune-action-plugin` for details about the sandboxing mechanism.
```

Comment:
The link name doesn't seem right. It's pointing to the sandboxing documentation so it should be called `sandboxing`

Reply:
This sentence is rendered as:

```
See Sandboxing for details about the sandboxing mechanism.
```

The link name is automatically selected by sphinx to be "sandboxing" from the section name the link is pointing to.

Reply:
Ack. The rendering is good, but it is still surprising when reading the raw rst files. We should use names that make sense to make these files easier to understand and work with.